service/Core/DataFormats/Office/MsExcelDecoder.cs

```diff
@@ -1,70 +1,59 @@
-    private readonly bool _withQuotes;
-    private readonly string _worksheetNumberTemplate;
-    private readonly string _endOfWorksheetMarkerTemplate;
-    private readonly string _rowPrefix;
-    private readonly string _columnSeparator;
-    private readonly string _rowSuffix;
-    private readonly string _blankCellValue;
-
-    public MsExcelDecoder(
-        bool withWorksheetNumber = true,
-        bool withEndOfWorksheetMarker = false,
-        bool withQuotes = true,
-        string? worksheetNumberTemplate = null,
-        string? endOfWorksheetMarkerTemplate = null,
-        string? rowPrefix = null,
-        string? columnSeparator = null,
-        string? rowSuffix = null,
-        string? blankCellValue = null)
-    {
-        this._withWorksheetNumber = withWorksheetNumber;
-        this._withEndOfWorksheetMarker = withEndOfWorksheetMarker;
-        this._withQuotes = withQuotes;
+    private readonly MsExcelConfig _config;
+    private readonly ILogger<MsExcelDecoder> _log;
 
-        this._worksheetNumberTemplate = worksheetNumberTemplate ?? DefaultSheetNumberTemplate;
-        this._endOfWorksheetMarkerTemplate = endOfWorksheetMarkerTemplate ?? DefaultEndOfSheetTemplate;
+    public IEnumerable<string> SupportedMimeTypes { get; } = [MimeTypes.MsExcelX, MimeTypes.MsExcel];
 
-        this._rowPrefix = rowPrefix ?? DefaultRowPrefix;
-        this._columnSeparator = columnSeparator ?? DefaultColumnSeparator;
-        this._rowSuffix = rowSuffix ?? DefaultRowSuffix;
-        this._blankCellValue = blankCellValue ?? DefaultBlankCellValue;
+    public MsExcelDecoder(MsExcelConfig? config = null, ILogger<MsExcelDecoder>? log = null)
+    {
+        this._config = config ?? new MsExcelConfig();
+        this._log = log ?? DefaultLogger<MsExcelDecoder>.Instance;
     }
 
-    public FileContent ExtractContent(string filename)
+    public Task<FileContent?> ExtractContentAsync(string handlerStepName, DataPipeline.FileDetails file, string filename, CancellationToken cancellationToken = default)
     {
         using var stream = File.OpenRead(filename);
-        return this.ExtractContent(stream);
+        return this.ExtractContentAsync(handlerStepName, file, stream, cancellationToken);
     }
 
-    public FileContent ExtractContent(BinaryData data)
+    public Task<FileContent?> ExtractContentAsync(string handlerStepName, DataPipeline.FileDetails file, BinaryData data, CancellationToken cancellationToken = default)
     {
         using var stream = data.ToStream();
-        return this.ExtractContent(stream);
+        return this.ExtractContentAsync(handlerStepName, file, stream, cancellationToken);
     }
 
-    public FileContent ExtractContent(Stream data)
+    public Task<FileContent?> ExtractContentAsync(string handlerStepName, DataPipeline.FileDetails file, Stream data, CancellationToken cancellationToken = default)
     {
+        if (file.MimeType == MimeTypes.MsExcel)
+        {
+            file.Log(
```

Comment:
I would remove this, e.g. decoders should not know anything about the pipeline, file details, and internal logging logic. If needed, we can include log data in FileContent, or return some log data as a separate object.

Reply:
I think that adding adding log data in `FileContent` could be a good solution. As we need this for logging the reason of  skipping the file, it could be a simple `SkippingReason` string property. Or do you have in mind something more structured?

Reply:
I would leave it open, a list of entries, reusing a struct/class similar to PipelineLogEntry, without using the same class though, at the cost of duplicating it. Something like:

```csharp
public class FileContent
{
    public sealed class LogEntry
    {
        [JsonPropertyOrder(0)]
        [JsonPropertyName("t")]
        public DateTimeOffset Time { get; set; } = DateTimeOffset.UtcNow;

        [JsonPropertyOrder(1)]
        [JsonPropertyName("src")]
        public string Source { get; set; }

        [JsonPropertyOrder(2)]
        [JsonPropertyName("txt")]
        public string Text { get; set; }

        public LogEntry(string source, string text)
        {
            this.Source = source;
            this.Text = text;
        }
    }
    
    [JsonPropertyOrder(0)]
    [JsonPropertyName("sections")]
    public List<FileSection> Sections { get; set; } = new();

    [JsonPropertyOrder(100)]
    [JsonPropertyName("log")]
    [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
    public List<LogEntry>? LogEntries { get; set; } = null;
}
```

Reply:
Alternatively, if there's just one log entry to add, I would just use the exception approach mentioned in the other comment, including in the exception all the details to be logged by the handler. That might be just easier and less code

Reply:
Yes, I think that using exception is the most common way to handle this situation.